components/brave_wallet/browser/internal/hd_key_ed25519.h

```diff
@@ -7,48 +7,63 @@
 #define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_ED25519_H_
 
 #include <memory>
+#include <optional>
 #include <string>
-#include <vector>
 
 #include "base/containers/span.h"
-#include "brave/components/brave_wallet/rust/lib.rs.h"
+#include "base/gtest_prod_util.h"
+#include "brave/components/brave_wallet/browser/internal/hd_key_utils.h"
 
 namespace brave_wallet {
 
-// This class implement basic EdDSA over ed25519 functionality of bip32-ed25519
+inline constexpr size_t kEd25519PrivateKeySize = 32;
+inline constexpr size_t kEd25519PublicKeySize = 32;
+inline constexpr size_t kEd25519KeypairSize =
+    kEd25519PrivateKeySize + kEd25519PublicKeySize;
+inline constexpr size_t kEd25519ChainCodeSize = 32;
+inline constexpr size_t kEd25519SignatureSize = 64;
+
+// This class implements basic EdDSA over ed25519 functionality of SLIP-0010
 // spec with 32 bytes private key and only allows private key derivation with
 // hardened index.
+// https://github.com/satoshilabs/slips/blob/master/slip-0010.md
 class HDKeyEd25519 {
  public:
-  HDKeyEd25519(std::string path,
-               rust::Box<Ed25519DalekExtendedSecretKeyResult>);
+  HDKeyEd25519();
   ~HDKeyEd25519();
   HDKeyEd25519(const HDKeyEd25519&) = delete;
   HDKeyEd25519& operator=(const HDKeyEd25519&) = delete;
 
-  static std::unique_ptr<HDKeyEd25519> GenerateFromSeed(
-      base::span<const uint8_t> seed);
-  static std::unique_ptr<HDKeyEd25519> GenerateFromPrivateKey(
-      base::span<const uint8_t> private_key);
+  static std::unique_ptr<HDKeyEd25519> GenerateFromSeedAndPath(
```

Comment:
any changes to the code should be a follow-up. We don't want to mix up replacing the ed25519 implementation with other changes to the code

Reply:
removed unrelated changes